scripthost/src/main/kotlin/com/commonwealthrobotics/bowlerkernel/scripthost/Session.kt

```diff
@@ -0,0 +1,271 @@
+                                    else -> null
+                                }
+
+                                val result = when {
+                                    requestId != null -> {
+                                        responseHandlers.remove(requestId)?.invoke(msg)
+                                        Either.Right(Unit)
+                                    }
+
+                                    msg.hasRunRequest() -> runRequest(msg.runRequest, sessionChannel)
+                                    msg.hasCancelRequest() -> TODO("Not yet implemented")
+                                    msg.hasNewConfig() -> TODO("Not yet implemented")
+                                    else -> throw IllegalStateException("Unhandled message type: $msg")
+                                }
+
+                                logger.debug { "Result of handling client message: $result" }
+
+                                if (msg.hasRunRequest()) {
+                                    if (result is Either.Left) {
+                                        logger.warn(result.a) { "Script did not run successfully." }
+                                        send(
+                                            sessionServerMessage {
+                                                errorBuilder.requestId = msg.runRequest.requestId
+                                                errorBuilder.description =
+                                                    "Script did not run successfully: ${result.a.localizedMessage}"
+                                            }
+                                        )
+                                    }
+                                }
+
+                                true
+                            }
+                        }
+                    }
+                }
+            } while (cont)
+        } catch (t: Throwable) {
+            logger.error(t) { "Unhandled error in session" }
+            throw t
+        } finally {
+            getKoin().unloadModules(modules)
+            logger.debug { "Session ended" }
+        }
+    }.consumeAsFlow()
+
+    init {
+        require(coroutineScope != GlobalScope) {
+            "A session may not be started in GlobalScope."
+        }
+
+        @Suppress("TooGenericExceptionCaught") // Okay because we just log and rethrow
+        try {
+            getKoin().loadModules(modules)
+        } catch (t: Throwable) {
+            logger.error(t) { "Unhandled error when constructing session" }
+            throw t
+        }
+    }
+
+    /**
+     * Handles a [RunRequest].
+     *
+     * @param runRequest The [RunRequest]
+     * @param sessionChannel A channel to send messages to the client.
+     * @return The result of loading and running the script.
+     */
+    private suspend fun runRequest(
+        runRequest: RunRequest,
+        sessionChannel: SendChannel<SessionServerMessage>
+    ): Either<Throwable, Any?> {
```

Comment:
`runRequest` needs to be asynchronous on some level, otherwise the main loop won't process anything while a script is running (defeating the point of it). IMO it should be completely asynchronous and have a `Unit` return type; it should be responsible for sending any responses that it needs to, including errors.

Reply:
Fixed